components/ai_chat/resources/untrusted_conversation_frame/components/markdown_renderer/index.tsx

```diff
@@ -330,9 +513,48 @@ export default function MarkdownRenderer(mainProps: MarkdownRendererProps) {
               disableLinkRestrictions={mainProps.disableLinkRestrictions}
             />
           ),
+          span: (props: any) => {
+            // Handle spans with fade-in class for new text
+            // (created by rehype plugin)
+            // The plugin wraps new text in spans with 'fade-in-new-text'
+            // class
+            const className = props.className
+            let hasFadeIn = false
+            if (className) {
+              if (typeof className === 'string') {
+                hasFadeIn = className.includes('fade-in-new-text')
+              } else if (Array.isArray(className)) {
+                hasFadeIn = className.includes('fade-in-new-text')
+              } else {
+                hasFadeIn = className === 'fade-in-new-text'
+              }
+            }
+            // Apply fade-in animation style if this span contains new text
+            const combinedClassName = hasFadeIn
+              ? `${props.className || ''} ${styles.fadeIn}`.trim()
+              : props.className
```

Comment:
I think you could add the styles to `:global(.fade-in-new-text)` and we wouldn't need to do this